### PR TITLE
Redshift data sharing - Make ShareObject.IAMRole a generic "Role"

### PR DESCRIPTION
**Pull request by dlpzx**
### Feature or Bugfix
- Feature
⚠️ merge after #1461 

### Detail
Complete design in https://github.com/data-dot-all/dataall/issues/955. This particular PR is part of the redshift dataset sharing implementation. The purpose of this particular PR is to make the create share object more generic. At the moment it is intertwined with IAM roles and share policies that need to be attached to them. 
- Rename ShareObject.principalIAMRoleName as ShareObject.principalRoleName and make it a mandatory field
- Rename usages of principalIAMRoleName
- Add migration script
- Make attachMissingPolicies an optional input parameter for createShareObject and default it to false
- Add principalRoleName as an optional input parameter for createShareObject 
- Modify createShareObject service to skip IAM actions on non Group/ConsumptionRole principal types

This PR fixes the creation of different types of share objects; but the ideal design should not include if-conditions in create_share_object. Instead the input could include a field for "additional_actions" such as ['CHECK_IAM_ROLE', 'ATTACH_MISSING_POLICIES'] that handle the different principal cases in a more generic way. I listed this as enhancements in the original ticket.

### Relates
- #955 

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dlpzx** [comment]
Testing
- Create share request for S3 dataset - stored principalRoleName is the IAM role (group or consumption role)
- Create share request for Redshift dataset - stored principalRoleName is the redshift role introduced
- Get principal in ShareView
- Migration script

File: frontend/src/modules/Catalog/components/RequestRedshiftAccessModal.js [hunk 1]
```diff
@@ -164,16 +164,13 @@ export const RequestRedshiftAccessModal = (props) => {
...
     let inputObject = {
       environmentUri: values.environmentUri,
       groupUri: values.groupUri,
-      principalId: principal,
-      principalType: type,
-      requestPurpose: values.comment,
-      attachMissingPolicies: values.attachMissingPolicies
+      principalId: values.connection,
+      principalRoleName: values.rsRole,
+      principalType: 'RedshiftRole',
```
Comment (noah-paige) [inline frontend/src/modules/Catalog/components/RequestRedshiftAccessModal.js:172]: nit: should this be pulled from `PrincipalType enum`

**Review by noah-paige — APPROVED** [review]
PR looks good - the one call out to make note of is the `def verify_principal_role` is used all over for submit, approve, verify, revoke shares (in shares_base) but currently would only work for IAM principals

leaving this as a call out so we do not forget when building out rest of redshift sharing mechanisms later in this PR series

File: frontend/src/modules/Catalog/components/RequestRedshiftAccessModal.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dlpzx) [reply frontend/src/modules/Catalog/components/RequestRedshiftAccessModal.js:172]: No, in this case they are just strings

**Comment by dlpzx** [comment]
> PR looks good - the one call out to make note of is the `def verify_principal_role` is used all over for submit, approve, verify, revoke shares (in shares_base) but currently would only work for IAM principals
> 
> leaving this as a call out so we do not forget when building out rest of redshift sharing mechanisms later in this PR series

Completely, I listed that as an action in the list of todos in the original issue

**Pull request merged**